packages/kbn-grid-layout/grid/grid_height_smoother.tsx

```diff
@@ -10,8 +10,12 @@
 import { css } from '@emotion/react';
 import React, { PropsWithChildren, useEffect, useRef } from 'react';
 import { combineLatest } from 'rxjs';
+import { euiThemeVars } from '@kbn/ui-theme';
 import { GridLayoutStateManager } from './types';
 
+const getViewportHeight = () =>
+  window.innerHeight || document.documentElement.clientHeight || document.body.clientHeight;
```

Comment:
I was looking for some eui hook but couldn't find anything else. Let me know if you find a way! 🙏🏼

Reply:
Still looking into this, sorry :bow:

Reply:
Testing this out some more, and I think we would be better off trying to accomplish this with flex 🤔 Right now, this isn't responding to the height of the viewport changing like it should:


https://github.com/user-attachments/assets/ede7a062-d2f8-4cb0-8d2d-4f245a291823

By doing it with flex instead (with something like `display: flex` on the parent, `flex: 1` to the row, and `height: 100%` to the panel)... I think we can avoid this? 🤔

Reply:
Thanks for spending the time on it and the suggestion, but I cannot make it work if I remove the manual height calculation. Is this something you've tested and it worked for you? Asking to make sure if it is possible and I am missing something obvious? 🙏🏼

Reply:
So I was able to get something working with setting the height to this instead: `calc(100vh - ${smoothHeightRefY}px - ${gutterSize}px)` - so with that, we can at least get rid of `getViewportHeight`

It's unfortunately still not a **great** solution, because in the actual Dashboard plugin, `smoothHeightRefY` changes as things load in the Dashboard (for example, the control group and unified search), which makes behaviour inconsistent. 

Maybe we proceed with the `calc` suggestion above, since it's at least a **tad** more reliable, and then I can figure out how to make this work a bit better for dashboards later :)

Reply:
Aaaah, I thought you achieved it with only applying the flex. I applied the changes you're suggested in the last comment and it is much better 😍 Thank you! I see no reason to use flex after all, unless I am missing something? 5134fcd663f85a9f5b0593042d9f72c8a5b4ae93
I see why it's not great though, maybe we should for now leave it as is, but submit an issue to target it once we are in the process of integrating a dashboard? Let me know, I am happy to submit an issue.

Reply:
I thought a little more about it and the last [commit](https://github.com/elastic/kibana/pull/200771/commits/e9590319bdeb7ac22d8e9557fdc558e09bef14b4) I submitted is an experiment I'd like you to look at. It uses the `fill-available` height option which is pretty well [supported](https://caniuse.com/?search=fill-available) (better than I expected). I added an element in the `app.tsx` file that pushes the content after 3 seconds after loading the page to test it out. What do you think? 🤔

Reply:
Oh!! The `fill-available` style is... really nice. I wonder if we could have a fallback for unsupported browsers? We would definitely want to add `width: -moz-available` at minimum... But I'm pretty confident that we could use this based on the "usage relative" graph 🤔 

![image](https://github.com/user-attachments/assets/d01bab2f-f74a-4874-8dc6-c7297251871e)

Reply:
😭 So I started testing it out in other browsers and looks like even though it is officially supported, it doesn't work the same, it does nothing there and only works in Chrome. After testing multiple options and combinations, I'll revert to the previous solution and create an issue about trying to fix it when we integrate in dashboards.

Reply:
Thanks for the thorough explanation!! The current implementation on Dashboard uses `flex`, so this seems like a good way forward :)